### PR TITLE
Kczernik/papp 34619

### PR DESCRIPTION
**Pull request by kczernik-splunk**
This PR is part of PAPP-34611 MS Graph Feature Parity with EWS it resolves the PAPP-34619 by adding Resolve name, Block sender and Unblock sender actions.   
  
All actions are described in this document: https://docs.google.com/document/d/1hVTXiaLzMZgbtlnYMjAu_8v61h6cwpR2HYLRC62oOlo/edit?usp=sharing   
  
While reviewing please comment on changes made in "office365_resolve_name.html" as it was my first time working in connectors html.  Test playbooks are now merged to next and cherry-picked to main, all tests are passing some actions unrelated to this merge required a rerun to succeed. Also please comment on verbose descriptions.

### CODE REVIEW

File: office365_connector.py [hunk 1]
```diff
@@ -3303,6 +2981,84 @@ def _handle_update_email(self, param):
...
+            ret_val_mailbox, response_mailbox = self._make_rest_call_helper(action_result, endpoint_mailbox)
+
+            self.save_progress(f"{ret_val} Got response {response}")
+            self.save_progress(f"{ret_val_proxy} Got response other mails {response_proxy}")
+            self.save_progress(f"{ret_val_address} Got response address {response_address}")
+            self.save_progress(f"{ret_val_mailbox} Got response mailbox {response_mailbox}")
+
+            action_result.add_data(response | response_proxy | response_address | (response_mailbox or {"userPurpose": None}))
+
+        # summary = action_result.update_summary({})
```
Comment (mposluszny-splunk) [inline office365_connector.py:3058]: remove commented out code

File: office365_connector.py [hunk 2]
```diff
@@ -3303,6 +2981,84 @@ def _handle_update_email(self, param):
...
+            endpoint_mailbox = f"/users/{user_id}/mailboxSettings/userPurpose"
+
+            ret_val_proxy, response_proxy = self._make_rest_call_helper(action_result, endpoint_other_mails)
+            ret_val_address, response_address = self._make_rest_call_helper(action_result, endpoint_other_address)
+            ret_val_mailbox, response_mailbox = self._make_rest_call_helper(action_result, endpoint_mailbox)
+
+            self.save_progress(f"{ret_val} Got response {response}")
+            self.save_progress(f"{ret_val_proxy} Got response other mails {response_proxy}")
+            self.save_progress(f"{ret_val_address} Got response address {response_address}")
+            self.save_progress(f"{ret_val_mailbox} Got response mailbox {response_mailbox}")
```
Comment (mposluszny-splunk) [inline office365_connector.py:3054]: should we print output of everything? Output may be large depending on response count, maybe debug_print or other would be better?

File: office365_connector.py (hunk 1 above)
Comment (kczernik-splunk) [reply office365_connector.py]: removed commented code

File: office365_connector.py (hunk 2 above)
Comment (kczernik-splunk) [reply office365_connector.py]: I have shortened the outputs, I believe due to nature of msgraph asset it is easier to debug and maintain action while using save_progress.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 repos:
 -   repo: https://github.com/phantomcyber/dev-cicd-tools
-    rev: v1.17
+    rev: v1.22
```
Comment (ivellios) [inline .pre-commit-config.yaml:3]: v1.23

**Review by ivellios — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 3 above)
Comment (ivellios) [inline .pre-commit-config.yaml:3]:
```suggestion
    rev: v1.23
```

Comment (kczernik-splunk) [reply .pre-commit-config.yaml:3]: fair point, on my way to rebuild wheels and rerun tests

**Review by mctonderski-splunk — APPROVED** [review]

**Pull request merged**